### ISSUE TITLE
False reports for CA1816 (Dispose methods should call SuppressFinalize)

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#1372]
#### Analyzer package

 Microsoft.NetCore.Analyzers

#### Analyzer

CallGCSuppressFinalizeCorrectlyAnalyzer

#### Repro steps

1. git clone https://github.com/JamesNK/Newtonsoft.Json
2. Install latest FxCopAnalyzers package https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.6.0-beta1-62315-01 to the solution
3. Filter down to CA1816 diagnostics, such as one reported in JsonReader.cs line 1101:

```
        void IDisposable.Dispose()
        {
            Dispose(true);
            GC.SuppressFinalize(this);
        }

```

#### Expected behavior
No CA1816
#### Actual behavior
```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CA1816	JsonReader.Dispose() calls GC.SuppressFinalize(object), a method that is typically only called within an implementation of 'IDisposable.Dispose'. Refer to the IDisposable pattern for more information.	Newtonsoft.Json(net40)	C:\Newtonsoft.Json\Src\Newtonsoft.Json\JsonReader.cs	1101	Active
```
We are actually invoking this inside interface implementation of `IDisposable.Dispose`, so it seems we are not correctly detecting analysis context. Likely an issue in `IMethodSymbolExtensions.IsDisposeImplementation`?

**Comment by 333fred** [comment]
That extension was short-circuiting by checking to see if the method was named "Dispose". When explicitly implementing, the name is actually `System.IDisposable.Dispose`.
